src/components/TreeGrid.ts

```diff
@@ -46,13 +46,65 @@ const wrapContent =
     state: 'closed',
   });
 
+const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<number, T[]>, domainOptionRoot: T): Row<T>[] => {
+  // initialize an array of rows
+  const rows: Row<T>[] = [];
+  const depth = 0;
+
+  const traverseHierarchy = (parent: T, depth: number) => {
+    // push the parent row
+    rows.push({
```

Comment:
Not sure what the new UIWG recommendations are for mutability but in the old conventions, mutability was avoided where possible. Could this be written in a non-mutable way, that constructs `rows` instead of modifying it?

Reply:
Sounds good!